x-pack/test/api_integration/apis/ml/annotations/get_annotations.ts

```diff
@@ -0,0 +1,167 @@
+      await ml.api.cleanMlIndices();
+    });
+
+    it('should fetch all annotations for jobId', async () => {
+      const requestBody = {
+        jobIds: [jobIds[0]],
+        earliestMs: 1454804100000,
+        latestMs: Date.now(),
+        maxAnnotations: 500,
+      };
+      const { body } = await supertest
+        .post('/api/ml/annotations')
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send(requestBody)
+        .expect(200);
+
+      expect(body.success).to.eql(true);
+      expect(body.annotations).not.to.be(undefined);
+      [jobIds[0]].forEach((jobId, idx) => {
+        expect(body.annotations).to.have.property(jobId);
+        expect(body.annotations[jobId]).to.have.length(1);
+
+        const indexedAnnotation = _.omit(body.annotations[jobId][0], '_id');
+        expect(indexedAnnotation).to.eql(testSetupAnnotations[idx]);
+      });
+    });
+
+    it('should fetch all annotations for multiple jobs', async () => {
+      const requestBody = {
+        jobIds: testSetupJobConfigs.map((j) => j.job_id),
+        earliestMs: 1454804100000,
+        latestMs: Date.now(),
+        maxAnnotations: 500,
+      };
+      const { body } = await supertest
+        .post('/api/ml/annotations')
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send(requestBody)
+        .expect(200);
+
+      expect(body.success).to.eql(true);
+      expect(body.annotations).not.to.be(undefined);
+      jobIds.forEach((jobId, idx) => {
+        expect(body.annotations).to.have.property(jobId);
+        expect(body.annotations[jobId]).to.have.length(1);
+
+        const indexedAnnotation = _.omit(body.annotations[jobId][0], '_id');
+        expect(indexedAnnotation).to.eql(testSetupAnnotations[idx]);
+      });
+    });
+
+    it('should fetch all annotations for user with viewer permissions', async () => {
+      const requestBody = {
+        jobIds: testSetupJobConfigs.map((j) => j.job_id),
+        earliestMs: 1454804100000,
+        latestMs: Date.now(),
+        maxAnnotations: 500,
+      };
+      const { body } = await supertest
+        .post('/api/ml/annotations')
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send(requestBody)
+        .expect(200);
+      expect(body.success).to.eql(true);
+      expect(body.annotations).not.to.be(undefined);
+      jobIds.forEach((jobId, idx) => {
+        expect(body.annotations).to.have.property(jobId);
+        expect(body.annotations[jobId]).to.have.length(1);
+
+        const indexedAnnotation = _.omit(body.annotations[jobId][0], '_id');
+        expect(indexedAnnotation).to.eql(testSetupAnnotations[idx]);
+      });
+    });
+
+    it('should not allow to fetch annotation for user with viewer permissions', async () => {
```

Comment:
```suggestion
    it('should not allow to fetch annotation for unauthorized user', async () => {
```

Reply:
Updated in https://github.com/elastic/kibana/pull/73068/commits/f45a447e78d38637d6649fcb92e6d3235b13c906